engine/storage/volume/src/org/apache/cloudstack/storage/volume/VolumeServiceImpl.java

```diff
@@ -2025,4 +2037,27 @@ public VolumeInfo updateHypervisorSnapshotReserveForVolume(DiskOffering diskOffe
 
         return volFactory.getVolume(volumeId);
     }
+
+    /**
+     *  inserts volume,snapshot details
+     * @param volumeId
+     * @param snapsShotid
+     */
+    public void insertVolumeDetailsVO(Long volumeId,Long snapsShotid)
```

Comment:
What is the need for this method? There is already a method addDetail() in VolumeDetailsDaoImpl.java. I would suggest checking the existing code to see what all are already available before implementing something.